src/scala/microsoft-spark-2.4.x/src/main/scala/org/apache/spark/api/dotnet/SerDe.scala

```diff
@@ -120,6 +136,15 @@ object SerDe {
     (0 until len).map(_ => readString(in)).toArray
   }
 
+  def readRowArr(in: DataInputStream): java.util.List[Row] = {
```

Comment:
You are trying to deserlialize the C# `IEnumerable<GenericRow>` and converting it to `java.util.List[Row]` because the SparkSession has the method signature `def createDataFrame(rows: java.util.List[Row], schema: StructType): DataFrame` .   Instead of doing that, let's return `Array[Row]` similar to the other SerDe methods.

From the C# side you can call a helper/proxy method that will 1) convert the `Array[Row]` to a `java.util.List[Row]` and 
2) call the SparkSession.createDataFrame .   I would add this helper method in `sql.api.dotnet.SQLUtils` .

@imback82  what is your thoughts about this ?

Reply:
I thought we were unable to find a static function that converted Array to java.util.List, or am I missing something?

Reply:
Made the changes, thanks for the suggestion!